exercises/concept/dancing-dots/.meta/design.md

```diff
@@ -0,0 +1,26 @@
+# Design
+
+## Learning objectives
+
+- Know how to define a behaviour with callbacks
+- Know how to provide a default implementation of a callback
+- Know how to use a behaviour
+- `@impl`
+
+## Out of scope
+
+- `@macrocallback`
+- `@optional_callbacks`
+
+## Concepts
+
+- `behaviours`
+- `use`
+
+## Prerequisites
+
+- `typespecs`
+- `structs`
+- `ast`
+- `enum`
+- `import`
```

Comment:
Is there a need for an analyzer as well?

The only thing I can thing of is that `DancingDots.Zoom.init/1` uses `Keyword` and `is_number` but it's very minor.

Reply:
I'm not sure if those two things even matter, using the access behaviour (`opts[:velocity]`) is also fine. I guess the only "bad idea" that technically works would be pattern matching on the keyword list. As for `is_number`, we could add more tests that would assert that it also allows floats or something, but I don't think that's really necessary.

I could think of two things, relevant to the concept:

[`362ceec` (#1103)](https://github.com/exercism/elixir/pull/1103/commits/362ceeca1b02e8256a11d383a404dd26d6bd9d6f)

Reply:
Those are better, yes :)